github/repos_tags.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// TagProtection represents a repository tag protection.
+type TagProtection struct {
+	ID      *int64  `json:"id"`
+	Pattern *string `json:"pattern"`
+}
+
+// TagProtectionRequest represents a request to create tag protection
+type TagProtectionRequest struct {
```

Comment:
Let's unexport this struct since it is only used internally.

```suggestion
type tagProtectionRequest struct {
```